main.cpp

```diff
@@ -42,29 +109,60 @@ struct Point
 template<typename Type>
 struct Wrapper
 {
-    Wrapper(Type&& t) : val(std::move(t)) 
-    { 
-        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    Wrapper() = default;
+
+    Wrapper(Type&& t) : val(std::move(t))
+    {
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl;
     }
+    
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+    
+private:
+    Type val;
 };
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& point) : val(std::move(point))
+    {
+        
+    }
 
-/*
- MAKE SURE YOU ARE NOT ON THE MASTER BRANCH
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val.toString() << ")" << std::endl;
+    }
 
- Commit your changes by clicking on the Source Control panel on the left, entering a message, and click [Commit and push].
- 
- If you didn't already: 
-    Make a pull request after you make your first commit
-    pin the pull request link and this repl.it link to our DM thread in a single message.
+private:
+    Point val;
+};
 
- send me a DM to review your pull request when the project is ready for review.
+template<typename T>
+void variadicHelper(T&& first)
+{
+    Wrapper<T>(std::forward<T>(first)).print();
+}
+
+void variadicHelper()
+{
+    
+}
+
+template<typename T, typename ...Args>
+void variadicHelper(T&& first, Args&& ... everythingElse)
+{
+    variadicHelper(std::forward<T>(first));
+
+    variadicHelper( std::forward<Args>(everythingElse) ... ); //recursive call
+}
```

Comment:
While this works, you don't need the empty-arg `variadicHelper`.  
You can have the single arg OR the no-arg versions of that function. 
Also, the instructions ask you to create the Wrapper in the templated multi-arg function, not in a separate function.
Can you explain in slack why you went with both the single-arg function and the no-arg function?

Reply:
Reply in Slack